connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -561,7 +563,7 @@ ErrorHandlingMetrics errorHandlingMetrics(ConnectorTaskId id) {
         // check if topic for dead letter queue exists
         String topic = connConfig.dlqTopicName();
         if (topic != null && !topic.isEmpty()) {
-            Map<String, Object> producerProps = producerConfigs(config);
+            Map<String, Object> producerProps = producerConfigs(id, config);
```

Comment:
Won't this result in the producer for the sink task error reporter to use `connector-producer-<id>`, which is more similar to _source_ connector client IDs (though there won't be an exact clash since connector names must be unique within a Connect cluster). Wouldn't it be better to have something more along the lines of `connector-error-<id>`? 

Unfortunately, there's no easy way to handle this without a third parameter in `producerConfigs(...)` -- which is not really that clean -- or adding a separate method that duplicates a lot of code -- which is bad.. Thoughts?

Reply:
Maybe this is fine: it is a producer, and it is used for the sink connector. Looking forward to feedback.

Reply:
Thanks @rhauch, this is a good point.  

As you say it _is_ a producer, so I think we should at least leave the word "producer" in the name to give that context to metrics inside the MBean. We could use "connector-error-producer-", or "connector-dlq-producer-".  While more vague, "connector-producer-" has the advantage of being less verbose and keeping the code a little cleaner.

Still, if desired we could handle switching names by externalizing the default and changing to `producerConfigs(String defaultClientId, WorkerConfig config)`, which is not quite as clean but OK I think.

Reply:
I do think having `connector-dlq-producer-<sink-connector-id>` would be much easier to correlate with how it's used. Passing in the String defaultClientId might not be horrible, or maybe passing in a clientIdPrefix as a third parameter. WDYT?

Reply:
Thanks @rhauch. `producerConfigs` currently only uses the `id` to build a default client id.  I think it's best to minimize what we pass to `producerConfigs`, so I've updated the PR with a `defaultClientId` parameter.  I'm using `connector-dlq-producer` for the DLQ default as suggested.